Build failure: php83-soap

## Steps To Reproduce

Steps to reproduce the behavior:

1. build `php83Extensions.soap` on master

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```
this derivation will be built:
  /nix/store/pcca7bcf6sckjfyq49cpxgcdwxfbj4d7-php-soap-8.3.13.drv
php-soap-x86_64-linux> building '/nix/store/pcca7bcf6sckjfyq49cpxgcdwxfbj4d7-php-soap-8.3.13.drv' on 'ssh://logi-nixos'
copying 0 paths
php-soap-x86_64-linux> Running phase: unpackPhase
php-soap-x86_64-linux> unpacking source archive /nix/store/9kp39cwlld91pry25ggan1c2ryxrpmv4-php-8.3.13.tar.bz2
php-soap-x86_64-linux> source root is php-8.3.13
php-soap-x86_64-linux> setting SOURCE_DATE_EPOCH to timestamp 1729622354 of file php-8.3.13/win32/wsyslog.c
php-soap-x86_64-linux> Running phase: patchPhase
php-soap-x86_64-linux> Running phase: genfiles
php-soap-x86_64-linux> genfiles: bison version 3.8.2 (ok)
php-soap-x86_64-linux> genfiles: re2c version 3.1 (ok)
php-soap-x86_64-linux> genfiles: Generating Zend parser and lexer files
php-soap-x86_64-linux> make: 'Zend/zend_language_parser.c' is up to date.
php-soap-x86_64-linux> make: 'Zend/zend_language_scanner.c' is up to date.
php-soap-x86_64-linux> make: 'Zend/zend_ini_parser.c' is up to date.
php-soap-x86_64-linux> make: 'Zend/zend_ini_scanner.c' is up to date.
php-soap-x86_64-linux> genfiles: Generating phpdbg parser and lexer files
php-soap-x86_64-linux> make: 'sapi/phpdbg/phpdbg_parser.c' is up to date.
php-soap-x86_64-linux> make: 'sapi/phpdbg/phpdbg_lexer.c' is up to date.
php-soap-x86_64-linux> genfiles: Generating json extension parser and lexer files
php-soap-x86_64-linux> make: 'ext/json/json_parser.tab.c' is up to date.
php-soap-x86_64-linux> make: 'ext/json/json_scanner.c' is up to date.
php-soap-x86_64-linux> genfiles: Generating PDO lexer file
php-soap-x86_64-linux> make: 'ext/pdo/pdo_sql_parser.c' is up to date.
php-soap-x86_64-linux> genfiles: Generating standard extension lexer files
php-soap-x86_64-linux> make: 'ext/standard/var_unserializer.c' is up to date.
php-soap-x86_64-linux> make: 'ext/standard/url_scanner_ex.c' is up to date.
php-soap-x86_64-linux> genfiles: Generating phar extension lexer file
php-soap-x86_64-linux> make: 'ext/phar/phar_path_check.c' is up to date.
php-soap-x86_64-linux> Running phase: cdToExtensionRootPhase
php-soap-x86_64-linux> Running phase: updateAutotoolsGnuConfigScriptsPhase
php-soap-x86_64-linux> Running phase: configurePhase
php-soap-x86_64-linux> Configuring for:
php-soap-x86_64-linux> PHP Api Version:         20230831
php-soap-x86_64-linux> Zend Module Api No:      20230831
php-soap-x86_64-linux> Zend Extension Api No:   420230831
php-soap-x86_64-linux> fixing libtool script ./build/ltmain.sh
php-soap-x86_64-linux> patching script interpreter paths in ./configure
php-soap-x86_64-linux> ./configure: interpreter directive changed from "#! /bin/sh" to "/nix/store/0irlcqx2n3qm6b1pc9rsd2i8qpvcccaj-bash-5.2p37/bin/sh"
php-soap-x86_64-linux> configure flags: --disable-static --prefix=/nix/store/myjs30rrrj1kpknl2g4kdq9z9x4i5rs4-php-soap-8.3.13 --bindir=/nix/store/myjs30rrrj1kpknl2g4kdq9z9x4i5rs4-php-soap-8.3.13/bin --sbindir=/nix/store/myjs30rrrj1kpknl2g4kdq9z9x4i5rs4-php-soap-8.3.13/sbin --includedir=/nix/store/q2s8s7vdndymmycf2viwpgsp19jpphjm-php-soap-8.3.13-dev/include --oldincludedir=/nix/store/q2s8s7vdndymmycf2viwpgsp19jpphjm-php-soap-8.3.13-dev/include --mandir=/nix/store/myjs30rrrj1kpknl2g4kdq9z9x4i5rs4-php-soap-8.3.13/share/man --infodir=/nix/store/myjs30rrrj1kpknl2g4kdq9z9x4i5rs4-php-soap-8.3.13/share/info --docdir=/nix/store/myjs30rrrj1kpknl2g4kdq9z9x4i5rs4-php-soap-8.3.13/share/doc/php-soap --libdir=/nix/store/myjs30rrrj1kpknl2g4kdq9z9x4i5rs4-php-soap-8.3.13/lib --libexecdir=/nix/store/myjs30rrrj1kpknl2g4kdq9z9x4i5rs4-php-soap-8.3.13/libexec --localedir=/nix/store/myjs30rrrj1kpknl2g4kdq9z9x4i5rs4-php-soap-8.3.13/share/locale --enable-soap
php-soap-x86_64-linux> checking for grep that handles long lines and -e... /nix/store/14aswfz6pm8zvyyy927xzmi8x4rnvlc8-gnugrep-3.11/bin/grep
php-soap-x86_64-linux> checking for egrep... /nix/store/14aswfz6pm8zvyyy927xzmi8x4rnvlc8-gnugrep-3.11/bin/grep -E
php-soap-x86_64-linux> checking for a sed that does not truncate output... /nix/store/gvivnz7rb2gz1vy7d9mdxgzj9didz7w2-gnused-4.9/bin/sed
php-soap-x86_64-linux> checking pkg-config is at least version 0.9.0... yes
php-soap-x86_64-linux> checking whether the C compiler works... yes
php-soap-x86_64-linux> checking for C compiler default output file name... a.out
php-soap-x86_64-linux> checking for suffix of executables...
php-soap-x86_64-linux> checking whether we are cross compiling... no
php-soap-x86_64-linux> checking for suffix of object files... o
php-soap-x86_64-linux> checking whether the compiler supports GNU C... yes
php-soap-x86_64-linux> checking whether gcc accepts -g... yes
php-soap-x86_64-linux> checking for gcc option to enable C11 features... none needed
php-soap-x86_64-linux> checking how to run the C preprocessor... gcc -E
php-soap-x86_64-linux> checking for egrep -e... (cached) /nix/store/14aswfz6pm8zvyyy927xzmi8x4rnvlc8-gnugrep-3.11/bin/grep -E
php-soap-x86_64-linux> checking for icc... no
php-soap-x86_64-linux> checking for suncc... no
php-soap-x86_64-linux> checking for system library directory... lib
php-soap-x86_64-linux> checking if compiler supports -Wl,-rpath,... yes
php-soap-x86_64-linux> checking build system type... x86_64-pc-linux-gnu
php-soap-x86_64-linux> checking host system type... x86_64-pc-linux-gnu
php-soap-x86_64-linux> checking target system type... x86_64-pc-linux-gnu
php-soap-x86_64-linux> checking for PHP prefix... /nix/store/cwiyd89mm650nx6nd3sz6jrs34wj4x7n-php-8.3.13
php-soap-x86_64-linux> checking for PHP includes... -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/main -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/TSRM -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/Zend -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/ext -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/ext/date/lib
php-soap-x86_64-linux> checking for PHP extension directory... /nix/store/cwiyd89mm650nx6nd3sz6jrs34wj4x7n-php-8.3.13/lib/php/extensions
php-soap-x86_64-linux> checking for PHP installed headers prefix... /nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php
php-soap-x86_64-linux> checking if debug is enabled... no
php-soap-x86_64-linux> checking if zts is enabled... no
php-soap-x86_64-linux> checking for gawk... gawk
php-soap-x86_64-linux> checking whether to enable SOAP support... yes, shared
php-soap-x86_64-linux> checking for libxml-2.0 >= 2.9.0... yes
php-soap-x86_64-linux> checking for a sed that does not truncate output... /nix/store/gvivnz7rb2gz1vy7d9mdxgzj9didz7w2-gnused-4.9/bin/sed
php-soap-x86_64-linux> checking for ld used by gcc... ld
php-soap-x86_64-linux> checking if the linker (ld) is GNU ld... yes
php-soap-x86_64-linux> checking for ld option to reload object files... -r
php-soap-x86_64-linux> checking for BSD-compatible nm... nm
php-soap-x86_64-linux> checking whether ln -s works... yes
php-soap-x86_64-linux> checking how to recognize dependent libraries... (cached) pass_all
php-soap-x86_64-linux> ./configure: line 5850: /usr/bin/file: No such file or directory
php-soap-x86_64-linux> checking for stdio.h... yes
php-soap-x86_64-linux> checking for stdlib.h... yes
php-soap-x86_64-linux> checking for string.h... yes
php-soap-x86_64-linux> checking for inttypes.h... yes
php-soap-x86_64-linux> checking for stdint.h... yes
php-soap-x86_64-linux> checking for strings.h... yes
php-soap-x86_64-linux> checking for sys/stat.h... yes
php-soap-x86_64-linux> checking for sys/types.h... yes
php-soap-x86_64-linux> checking for unistd.h... yes
php-soap-x86_64-linux> checking for dlfcn.h... yes
php-soap-x86_64-linux> checking the maximum length of command line arguments... 1572864
php-soap-x86_64-linux> checking command to parse nm output from gcc object... ok
php-soap-x86_64-linux> checking for objdir... .libs
php-soap-x86_64-linux> checking for ar... ar
php-soap-x86_64-linux> checking for ranlib... ranlib
php-soap-x86_64-linux> checking for strip... strip
php-soap-x86_64-linux> checking if gcc supports -fno-rtti -fno-exceptions... no
php-soap-x86_64-linux> checking for gcc option to produce PIC... -fPIC
php-soap-x86_64-linux> checking if gcc PIC flag -fPIC works... yes
php-soap-x86_64-linux> checking if gcc static flag -static works... no
php-soap-x86_64-linux> checking if gcc supports -c -o file.o... yes
php-soap-x86_64-linux> checking whether the gcc linker (ld) supports shared libraries... yes
php-soap-x86_64-linux> checking whether -lc should be explicitly linked in... no
php-soap-x86_64-linux> checking dynamic linker characteristics... GNU/Linux ld.so
php-soap-x86_64-linux> checking how to hardcode library paths into programs... immediate
php-soap-x86_64-linux> checking whether stripping libraries is possible... yes
php-soap-x86_64-linux> checking if libtool supports shared libraries... yes
php-soap-x86_64-linux> checking whether to build shared libraries... yes
php-soap-x86_64-linux> checking whether to build static libraries... no
php-soap-x86_64-linux>
php-soap-x86_64-linux> creating libtool
php-soap-x86_64-linux> appending configuration tag "CXX" to libtool
php-soap-x86_64-linux> checking whether the g++ linker (ld) supports shared libraries... yes
php-soap-x86_64-linux> checking for g++ option to produce PIC...
php-soap-x86_64-linux> checking if g++ static flag  works... yes
php-soap-x86_64-linux> checking if g++ supports -c -o file.o... yes
php-soap-x86_64-linux> checking whether the g++ linker (ld) supports shared libraries... yes
php-soap-x86_64-linux> checking dynamic linker characteristics... GNU/Linux ld.so
php-soap-x86_64-linux> (cached) (cached) checking how to hardcode library paths into programs... unsupported
php-soap-x86_64-linux> configure: patching config.h.in
php-soap-x86_64-linux> configure: creating ./config.status
php-soap-x86_64-linux> config.status: creating config.h
php-soap-x86_64-linux> Running phase: buildPhase
php-soap-x86_64-linux> build flags: -j30 SHELL=/nix/store/0irlcqx2n3qm6b1pc9rsd2i8qpvcccaj-bash-5.2p37/bin/bash
php-soap-x86_64-linux> /nix/store/0irlcqx2n3qm6b1pc9rsd2i8qpvcccaj-bash-5.2p37/bin/bash /build/php-8.3.13/ext/soap/libtool --tag=CC --mode=compile gcc -I. -I/build/php-8.3.13/ext/soap -I/build/php-8.3.13/ext/soap/include -I/build/php-8.3.13/ext/soap/main -I/build/php-8.3.13/ext/soap -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/main -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/TSRM -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/Zend -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/ext -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/ext/date/lib -I/nix/store/ixsyzzg90q0p20svkwl1nhkf24axvpcn-libxml2-2.13.4-dev/include/libxml2  -DHAVE_CONFIG_H  -g -O2 -D_GNU_SOURCE   -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 -DZEND_COMPILE_DL_EXT=1 -c /build/php-8.3.13/ext/soap/soap.c -o soap.lo  -MMD -MF soap.dep -MT soap.lo
php-soap-x86_64-linux> /nix/store/0irlcqx2n3qm6b1pc9rsd2i8qpvcccaj-bash-5.2p37/bin/bash /build/php-8.3.13/ext/soap/libtool --tag=CC --mode=compile gcc -I. -I/build/php-8.3.13/ext/soap -I/build/php-8.3.13/ext/soap/include -I/build/php-8.3.13/ext/soap/main -I/build/php-8.3.13/ext/soap -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/main -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/TSRM -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/Zend -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/ext -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/ext/date/lib -I/nix/store/ixsyzzg90q0p20svkwl1nhkf24axvpcn-libxml2-2.13.4-dev/include/libxml2  -DHAVE_CONFIG_H  -g -O2 -D_GNU_SOURCE   -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 -DZEND_COMPILE_DL_EXT=1 -c /build/php-8.3.13/ext/soap/php_encoding.c -o php_encoding.lo  -MMD -MF php_encoding.dep -MT php_encoding.lo
php-soap-x86_64-linux> /nix/store/0irlcqx2n3qm6b1pc9rsd2i8qpvcccaj-bash-5.2p37/bin/bash /build/php-8.3.13/ext/soap/libtool --tag=CC --mode=compile gcc -I. -I/build/php-8.3.13/ext/soap -I/build/php-8.3.13/ext/soap/include -I/build/php-8.3.13/ext/soap/main -I/build/php-8.3.13/ext/soap -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/main -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/TSRM -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/Zend -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/ext -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/ext/date/lib -I/nix/store/ixsyzzg90q0p20svkwl1nhkf24axvpcn-libxml2-2.13.4-dev/include/libxml2  -DHAVE_CONFIG_H  -g -O2 -D_GNU_SOURCE   -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 -DZEND_COMPILE_DL_EXT=1 -c /build/php-8.3.13/ext/soap/php_http.c -o php_http.lo  -MMD -MF php_http.dep -MT php_http.lo
php-soap-x86_64-linux> /nix/store/0irlcqx2n3qm6b1pc9rsd2i8qpvcccaj-bash-5.2p37/bin/bash /build/php-8.3.13/ext/soap/libtool --tag=CC --mode=compile gcc -I. -I/build/php-8.3.13/ext/soap -I/build/php-8.3.13/ext/soap/include -I/build/php-8.3.13/ext/soap/main -I/build/php-8.3.13/ext/soap -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/main -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/TSRM -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/Zend -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/ext -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/ext/date/lib -I/nix/store/ixsyzzg90q0p20svkwl1nhkf24axvpcn-libxml2-2.13.4-dev/include/libxml2  -DHAVE_CONFIG_H  -g -O2 -D_GNU_SOURCE   -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 -DZEND_COMPILE_DL_EXT=1 -c /build/php-8.3.13/ext/soap/php_packet_soap.c -o php_packet_soap.lo  -MMD -MF php_packet_soap.dep -MT php_packet_soap.lo
php-soap-x86_64-linux> /nix/store/0irlcqx2n3qm6b1pc9rsd2i8qpvcccaj-bash-5.2p37/bin/bash /build/php-8.3.13/ext/soap/libtool --tag=CC --mode=compile gcc -I. -I/build/php-8.3.13/ext/soap -I/build/php-8.3.13/ext/soap/include -I/build/php-8.3.13/ext/soap/main -I/build/php-8.3.13/ext/soap -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/main -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/TSRM -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/Zend -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/ext -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/ext/date/lib -I/nix/store/ixsyzzg90q0p20svkwl1nhkf24axvpcn-libxml2-2.13.4-dev/include/libxml2  -DHAVE_CONFIG_H  -g -O2 -D_GNU_SOURCE   -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 -DZEND_COMPILE_DL_EXT=1 -c /build/php-8.3.13/ext/soap/php_schema.c -o php_schema.lo  -MMD -MF php_schema.dep -MT php_schema.lo
php-soap-x86_64-linux> /nix/store/0irlcqx2n3qm6b1pc9rsd2i8qpvcccaj-bash-5.2p37/bin/bash /build/php-8.3.13/ext/soap/libtool --tag=CC --mode=compile gcc -I. -I/build/php-8.3.13/ext/soap -I/build/php-8.3.13/ext/soap/include -I/build/php-8.3.13/ext/soap/main -I/build/php-8.3.13/ext/soap -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/main -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/TSRM -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/Zend -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/ext -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/ext/date/lib -I/nix/store/ixsyzzg90q0p20svkwl1nhkf24axvpcn-libxml2-2.13.4-dev/include/libxml2  -DHAVE_CONFIG_H  -g -O2 -D_GNU_SOURCE   -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 -DZEND_COMPILE_DL_EXT=1 -c /build/php-8.3.13/ext/soap/php_sdl.c -o php_sdl.lo  -MMD -MF php_sdl.dep -MT php_sdl.lo
php-soap-x86_64-linux> /nix/store/0irlcqx2n3qm6b1pc9rsd2i8qpvcccaj-bash-5.2p37/bin/bash /build/php-8.3.13/ext/soap/libtool --tag=CC --mode=compile gcc -I. -I/build/php-8.3.13/ext/soap -I/build/php-8.3.13/ext/soap/include -I/build/php-8.3.13/ext/soap/main -I/build/php-8.3.13/ext/soap -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/main -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/TSRM -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/Zend -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/ext -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/ext/date/lib -I/nix/store/ixsyzzg90q0p20svkwl1nhkf24axvpcn-libxml2-2.13.4-dev/include/libxml2  -DHAVE_CONFIG_H  -g -O2 -D_GNU_SOURCE   -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 -DZEND_COMPILE_DL_EXT=1 -c /build/php-8.3.13/ext/soap/php_xml.c -o php_xml.lo  -MMD -MF php_xml.dep -MT php_xml.lo
php-soap-x86_64-linux> mkdir .libs
php-soap-x86_64-linux>  gcc -I. -I/build/php-8.3.13/ext/soap -I/build/php-8.3.13/ext/soap/include -I/build/php-8.3.13/ext/soap/main -I/build/php-8.3.13/ext/soap -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/main -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/TSRM -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/Zend -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/ext -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/ext/date/lib -I/nix/store/ixsyzzg90q0p20svkwl1nhkf24axvpcn-libxml2-2.13.4-dev/include/libxml2 -DHAVE_CONFIG_H -g -O2 -D_GNU_SOURCE -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 -DZEND_COMPILE_DL_EXT=1 -c /build/php-8.3.13/ext/soap/php_encoding.c -MMD -MF php_encoding.dep -MT php_encoding.lo  -fPIC -DPIC -o .libs/php_encoding.o
php-soap-x86_64-linux>  gcc -I. -I/build/php-8.3.13/ext/soap -I/build/php-8.3.13/ext/soap/include -I/build/php-8.3.13/ext/soap/main -I/build/php-8.3.13/ext/soap -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/main -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/TSRM -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/Zend -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/ext -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/ext/date/lib -I/nix/store/ixsyzzg90q0p20svkwl1nhkf24axvpcn-libxml2-2.13.4-dev/include/libxml2 -DHAVE_CONFIG_H -g -O2 -D_GNU_SOURCE -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 -DZEND_COMPILE_DL_EXT=1 -c /build/php-8.3.13/ext/soap/php_http.c -MMD -MF php_http.dep -MT php_http.lo  -fPIC -DPIC -o .libs/php_http.o
php-soap-x86_64-linux>  gcc -I. -I/build/php-8.3.13/ext/soap -I/build/php-8.3.13/ext/soap/include -I/build/php-8.3.13/ext/soap/main -I/build/php-8.3.13/ext/soap -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/main -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/TSRM -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/Zend -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/ext -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/ext/date/lib -I/nix/store/ixsyzzg90q0p20svkwl1nhkf24axvpcn-libxml2-2.13.4-dev/include/libxml2 -DHAVE_CONFIG_H -g -O2 -D_GNU_SOURCE -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 -DZEND_COMPILE_DL_EXT=1 -c /build/php-8.3.13/ext/soap/php_sdl.c -MMD -MF php_sdl.dep -MT php_sdl.lo  -fPIC -DPIC -o .libs/php_sdl.o
php-soap-x86_64-linux>  gcc -I. -I/build/php-8.3.13/ext/soap -I/build/php-8.3.13/ext/soap/include -I/build/php-8.3.13/ext/soap/main -I/build/php-8.3.13/ext/soap -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/main -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/TSRM -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/Zend -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/ext -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/ext/date/lib -I/nix/store/ixsyzzg90q0p20svkwl1nhkf24axvpcn-libxml2-2.13.4-dev/include/libxml2 -DHAVE_CONFIG_H -g -O2 -D_GNU_SOURCE -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 -DZEND_COMPILE_DL_EXT=1 -c /build/php-8.3.13/ext/soap/soap.c -MMD -MF soap.dep -MT soap.lo  -fPIC -DPIC -o .libs/soap.o
php-soap-x86_64-linux>  gcc -I. -I/build/php-8.3.13/ext/soap -I/build/php-8.3.13/ext/soap/include -I/build/php-8.3.13/ext/soap/main -I/build/php-8.3.13/ext/soap -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/main -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/TSRM -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/Zend -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/ext -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/ext/date/lib -I/nix/store/ixsyzzg90q0p20svkwl1nhkf24axvpcn-libxml2-2.13.4-dev/include/libxml2 -DHAVE_CONFIG_H -g -O2 -D_GNU_SOURCE -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 -DZEND_COMPILE_DL_EXT=1 -c /build/php-8.3.13/ext/soap/php_packet_soap.c -MMD -MF php_packet_soap.dep -MT php_packet_soap.lo  -fPIC -DPIC -o .libs/php_packet_soap.o
php-soap-x86_64-linux>  gcc -I. -I/build/php-8.3.13/ext/soap -I/build/php-8.3.13/ext/soap/include -I/build/php-8.3.13/ext/soap/main -I/build/php-8.3.13/ext/soap -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/main -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/TSRM -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/Zend -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/ext -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/ext/date/lib -I/nix/store/ixsyzzg90q0p20svkwl1nhkf24axvpcn-libxml2-2.13.4-dev/include/libxml2 -DHAVE_CONFIG_H -g -O2 -D_GNU_SOURCE -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 -DZEND_COMPILE_DL_EXT=1 -c /build/php-8.3.13/ext/soap/php_xml.c -MMD -MF php_xml.dep -MT php_xml.lo  -fPIC -DPIC -o .libs/php_xml.o
php-soap-x86_64-linux>  gcc -I. -I/build/php-8.3.13/ext/soap -I/build/php-8.3.13/ext/soap/include -I/build/php-8.3.13/ext/soap/main -I/build/php-8.3.13/ext/soap -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/main -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/TSRM -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/Zend -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/ext -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/ext/date/lib -I/nix/store/ixsyzzg90q0p20svkwl1nhkf24axvpcn-libxml2-2.13.4-dev/include/libxml2 -DHAVE_CONFIG_H -g -O2 -D_GNU_SOURCE -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 -DZEND_COMPILE_DL_EXT=1 -c /build/php-8.3.13/ext/soap/php_schema.c -MMD -MF php_schema.dep -MT php_schema.lo  -fPIC -DPIC -o .libs/php_schema.o
php-soap-x86_64-linux> /nix/store/0irlcqx2n3qm6b1pc9rsd2i8qpvcccaj-bash-5.2p37/bin/bash /build/php-8.3.13/ext/soap/libtool --tag=CC --mode=link gcc -shared -I/build/php-8.3.13/ext/soap/include -I/build/php-8.3.13/ext/soap/main -I/build/php-8.3.13/ext/soap -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/main -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/TSRM -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/Zend -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/ext -I/nix/store/d52vyb6i74wvn9yw6c2nvwz75lxs606r-php-8.3.13-dev/include/php/ext/date/lib -I/nix/store/ixsyzzg90q0p20svkwl1nhkf24axvpcn-libxml2-2.13.4-dev/include/libxml2  -DHAVE_CONFIG_H  -g -O2 -D_GNU_SOURCE    -o soap.la -export-dynamic -avoid-version -prefer-pic -module -rpath /build/php-8.3.13/ext/soap/modules  soap.lo php_encoding.lo php_http.lo php_packet_soap.lo php_schema.lo php_sdl.lo php_xml.lo -Wl,-rpath,/nix/store/067800yh33y5m156i6rhmzhsn8wxbbnh-libxml2-2.13.4/lib -L/nix/store/067800yh33y5m156i6rhmzhsn8wxbbnh-libxml2-2.13.4/lib -lxml2
php-soap-x86_64-linux> gcc -shared  .libs/soap.o .libs/php_encoding.o .libs/php_http.o .libs/php_packet_soap.o .libs/php_schema.o .libs/php_sdl.o .libs/php_xml.o  -Wl,--rpath -Wl,/nix/store/067800yh33y5m156i6rhmzhsn8wxbbnh-libxml2-2.13.4/lib -Wl,--rpath -Wl,/nix/store/067800yh33y5m156i6rhmzhsn8wxbbnh-libxml2-2.13.4/lib -L/nix/store/067800yh33y5m156i6rhmzhsn8wxbbnh-libxml2-2.13.4/lib /nix/store/067800yh33y5m156i6rhmzhsn8wxbbnh-libxml2-2.13.4/lib/libxml2.so  -Wl,-rpath -Wl,/nix/store/067800yh33y5m156i6rhmzhsn8wxbbnh-libxml2-2.13.4/lib -Wl,-soname -Wl,soap.so -o .libs/soap.so
php-soap-x86_64-linux> creating soap.la
php-soap-x86_64-linux> (cd .libs && rm -f soap.la && ln -s ../soap.la soap.la)
php-soap-x86_64-linux> /nix/store/0irlcqx2n3qm6b1pc9rsd2i8qpvcccaj-bash-5.2p37/bin/bash /build/php-8.3.13/ext/soap/libtool --tag=CC --mode=install cp ./soap.la /build/php-8.3.13/ext/soap/modules
php-soap-x86_64-linux> cp ./.libs/soap.so /build/php-8.3.13/ext/soap/modules/soap.so
php-soap-x86_64-linux> cp ./.libs/soap.lai /build/php-8.3.13/ext/soap/modules/soap.la
php-soap-x86_64-linux> PATH="$PATH:/sbin" ldconfig -n /build/php-8.3.13/ext/soap/modules
php-soap-x86_64-linux> ----------------------------------------------------------------------
php-soap-x86_64-linux> Libraries have been installed in:
php-soap-x86_64-linux>    /build/php-8.3.13/ext/soap/modules
php-soap-x86_64-linux>
php-soap-x86_64-linux> If you ever happen to want to link against installed libraries
php-soap-x86_64-linux> in a given directory, LIBDIR, you must either use libtool, and
php-soap-x86_64-linux> specify the full pathname of the library, or use the `-LLIBDIR'
php-soap-x86_64-linux> flag during linking and do at least one of the following:
php-soap-x86_64-linux>    - add LIBDIR to the `LD_LIBRARY_PATH' environment variable
php-soap-x86_64-linux>      during execution
php-soap-x86_64-linux>    - add LIBDIR to the `LD_RUN_PATH' environment variable
php-soap-x86_64-linux>      during linking
php-soap-x86_64-linux>    - use the `-Wl,--rpath -Wl,LIBDIR' linker flag
php-soap-x86_64-linux>
php-soap-x86_64-linux> See any operating system documentation about shared libraries for
php-soap-x86_64-linux> more information, such as the ld(1) and ld.so(8) manual pages.
php-soap-x86_64-linux> ----------------------------------------------------------------------
php-soap-x86_64-linux>
php-soap-x86_64-linux> Build complete.
php-soap-x86_64-linux> Don't forget to run 'make test'.
php-soap-x86_64-linux>
php-soap-x86_64-linux> Running phase: checkPhase
php-soap-x86_64-linux>
php-soap-x86_64-linux> Build complete.
php-soap-x86_64-linux> Don't forget to run 'make test'.
php-soap-x86_64-linux>
php-soap-x86_64-linux>
php-soap-x86_64-linux> =====================================================================
php-soap-x86_64-linux> PHP         : /nix/store/cwiyd89mm650nx6nd3sz6jrs34wj4x7n-php-8.3.13/bin/php
php-soap-x86_64-linux> PHP_SAPI    : cli
php-soap-x86_64-linux> PHP_VERSION : 8.3.13
php-soap-x86_64-linux> ZEND_VERSION: 4.3.13
php-soap-x86_64-linux> PHP_OS      : Linux - Linux localhost 6.6.60 #1-NixOS SMP PREEMPT_DYNAMIC Fri Nov  8 15:28:28 UTC 2024 x86_64
php-soap-x86_64-linux> INI actual  : /build/php-8.3.13/ext/soap/tmp-php.ini
php-soap-x86_64-linux> More .INIs  :
php-soap-x86_64-linux> ---------------------------------------------------------------------
php-soap-x86_64-linux> PHP         : /nix/store/cwiyd89mm650nx6nd3sz6jrs34wj4x7n-php-8.3.13/bin/php-cgi
php-soap-x86_64-linux> PHP_SAPI    : cgi-fcgi
php-soap-x86_64-linux> PHP_VERSION : 8.3.13
php-soap-x86_64-linux> ZEND_VERSION: 4.3.13
php-soap-x86_64-linux> PHP_OS      : Linux - Linux localhost 6.6.60 #1-NixOS SMP PREEMPT_DYNAMIC Fri Nov  8 15:28:28 UTC 2024 x86_64
php-soap-x86_64-linux> INI actual  : /build/php-8.3.13/ext/soap/tmp-php.ini
php-soap-x86_64-linux> More .INIs  :
php-soap-x86_64-linux> ---------------------------------------------------------------------
php-soap-x86_64-linux> ---------------------------------------------------------------------
php-soap-x86_64-linux> PHP         : /nix/store/cwiyd89mm650nx6nd3sz6jrs34wj4x7n-php-8.3.13/bin/phpdbg
php-soap-x86_64-linux> PHP_SAPI    : phpdbg
php-soap-x86_64-linux> PHP_VERSION : 8.3.13
php-soap-x86_64-linux> ZEND_VERSION: 4.3.13
php-soap-x86_64-linux> PHP_OS      : Linux - Linux localhost 6.6.60 #1-NixOS SMP PREEMPT_DYNAMIC Fri Nov  8 15:28:28 UTC 2024 x86_64
php-soap-x86_64-linux> INI actual  : /build/php-8.3.13/ext/soap/tmp-php.ini
php-soap-x86_64-linux> More .INIs  :
php-soap-x86_64-linux> ---------------------------------------------------------------------
php-soap-x86_64-linux> CWD         : /build/php-8.3.13/ext/soap
php-soap-x86_64-linux> Extra dirs  :
php-soap-x86_64-linux> VALGRIND    : Not used
php-soap-x86_64-linux> =====================================================================
php-soap-x86_64-linux> TIME START 2024-11-22 16:43:15
php-soap-x86_64-linux> =====================================================================
php-soap-x86_64-linux> PASS GH-14586: SoapFault::__construct() leaks memory if called twice [tests/SoapFault/gh14586.phpt]
php-soap-x86_64-linux> PASS SOAP handling of <any> [tests/any.phpt]
[...]
php-soap-x86_64-linux> FAIL GH-15711 (SoapClient can't convert BackedEnum to scalar value) [tests/gh15711.phpt]
[...]
php-soap-x86_64-linux> =====================================================================
php-soap-x86_64-linux> TIME END 2024-11-22 16:43:27
php-soap-x86_64-linux>
php-soap-x86_64-linux> =====================================================================
php-soap-x86_64-linux> TEST RESULT SUMMARY
php-soap-x86_64-linux> ---------------------------------------------------------------------
php-soap-x86_64-linux> Exts skipped    :     1 (session)
php-soap-x86_64-linux> Exts tested     :    12
php-soap-x86_64-linux> ---------------------------------------------------------------------
php-soap-x86_64-linux>
php-soap-x86_64-linux> Number of tests :   554               538
php-soap-x86_64-linux> Tests skipped   :    16 (  2.9%) --------
php-soap-x86_64-linux> Tests warned    :     0 (  0.0%) (  0.0%)
php-soap-x86_64-linux> Tests failed    :     1 (  0.2%) (  0.2%)
php-soap-x86_64-linux> Tests passed    :   537 ( 96.9%) ( 99.8%)
php-soap-x86_64-linux> ---------------------------------------------------------------------
php-soap-x86_64-linux> Time taken      :    12 seconds
php-soap-x86_64-linux> =====================================================================
php-soap-x86_64-linux>
php-soap-x86_64-linux> =====================================================================
php-soap-x86_64-linux> FAILED TEST SUMMARY
php-soap-x86_64-linux> ---------------------------------------------------------------------
php-soap-x86_64-linux> GH-15711 (SoapClient can't convert BackedEnum to scalar value) [tests/gh15711.phpt]
php-soap-x86_64-linux> =====================================================================
php-soap-x86_64-linux> make: *** [Makefile:136: test] Error 1
```

</details>

## Additional context

Maybe related the removal of the patch fixing this test here https://github.com/NixOS/nixpkgs/pull/357904

## Metadata

- system: `"aarch64-darwin"`
 - host os: `Darwin 24.1.0, macOS 15.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.24.10`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/store/5q79468yd486p049jk147djamjz9v0nv-source`

But build remotely on an `x86_64-linux`

## Notify maintainers

@NixOS/php @pan93412 

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/top-level/php-packages.nix
+++ b/pkgs/top-level/php-packages.nix
@@ -684,6 +684,13 @@ in {
               url = "https://github.com/php/php-src/commit/4fe821311cafb18ca8bdf20b9d796c48a13ba552.patch";
               hash = "sha256-YC3I0BQi3o3+VmRu/UqpqPpaSC+ekPqzbORTHftbPvY=";
             })
+          ]
+          ++ lib.optionals (lib.versionAtLeast php.version "8.3" && lib.versionOlder php.version "8.4") [
+            # https://github.com/php/php-src/pull/16733 (fix soap test)
+            (fetchpatch {
+              url = "https://github.com/php/php-src/commit/5c308d61db104854e4ff84ab123e3ea56e1b4046.patch";
+              hash = "sha256-xQ4Sg4kL0cgHYauRW2AzGgFXfcqtxeRVhI9zNh7CsoM=";
+            })
           ];
         }
         {

